### PR TITLE
Fixed getting multiple events from socket

### LINKED ISSUE #147: Bug: First Click On "Next Round"

### LINKED ISSUE #149: Duplicating on Going to other Event
-Not Cleaning Something up 
-Check on end Game

### PR DESCRIPTION
**Pull request by kevslim**
There was a weird issue where when using removeAllListeners([EVENT_NAMES]) with the parameter being an array of the EVENT_NAMES to remove, it didn't do anything. I tried commenting it out and running it again, and the same behavior was occurring which led me to believe that removeAllListeners([EVENT_NAMES]) was not doing anything. I then just tried removeAllListeners() without a parameter and the socket events were not duplicating.

I also did a quick and dirty fix to clicking the next round button twice for the first time. I just set the initial redux store value of the round number to be 2, since we manually set the first round to 1 when we initialize the game.

Closes #147 Closes #149 

So apparently, the removeAllListeners() in the componentDidMount is needed with the parameters of all the EVENT_NAMES, AND the empty parameter removeAllListeners() is also needed in the componentDidUnmount. I'm not sure why...

### CODE REVIEW

**Comment by kevslim** [comment]
Wait don't approve yet, maybe need to fix something else

File: client/components/controllers/Controller.js [hunk 1]
```diff
@@ -87,8 +93,8 @@ class Controller extends Component {
   }
 
   componentWillUnmount() {
+    socket.removeAllListeners()
```
Comment (pckilgore) [inline client/components/controllers/Controller.js]: DId you ever add  a console log here to see if it actually gets called?  Maybe all our assumptions are wrong...

File: client/store/event.js [hunk 2]
```diff
@@ -158,7 +158,7 @@ export default function(state = defaultEvents, action) {
     case RESET_ROUNDS: {
       return {
         byId: {...state.byId},
-        round: 1
+        round: 2
```
Comment (pckilgore) [inline client/store/event.js]: I guess I don't understand how this works ? It feels fragile especially since we should be saving this stuff in the DB.

**Comment by pckilgore** [comment]
btw only reason I rejected this was abecause you said so...

File: client/store/event.js (hunk 2 above)
Comment (kevslim) [reply client/store/event.js]: It's a temporary fix while I get modify the Event model

**Pull request merged**